master/buildbot/asyncio.py

```diff
@@ -0,0 +1,111 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Alternatively, you can use and copy this module under the MIT License
+# Copyright Buildbot Team Members
+
+import asyncio
+import inspect
+import sys
+from asyncio import base_events
+from asyncio import events
+
+from twisted.internet import defer
+
+
+def deferred_await(self):
+    # if a deferred is awaited from a asyncio loop context, we must return
+    # the future wrapper, but if it is awaited from normal twisted loop
+    # we must return self.
+    if isinstance(asyncio.get_event_loop(), TwistedLoop):
+        return self.asFuture(asyncio.get_event_loop())
+    return self
+
+
+defer.Deferred.__await__ = deferred_await
+
+
+def as_deferred(f):
+    return asyncio.get_event_loop().as_deferred(f)
+
+
+if sys.version_info[:2] >= (3, 7):
+    def make_handle(callback, args, loop, context):
+        return events.Handle(callback, args, loop, context)
+else:
```

Comment:
Since there's a flake8 failure anyway, I think it makes sense to mark this else with `pragma: no cover`

Reply:
(though I see that we should run coverage on 3.6 too, it's interesting why this wasn't covered.

Reply:
its because I didn't put the ci/ prefix before coverage. It is required so that codecov wait for the end of the test before sending results